### ISSUE TITLE
Predicate for Object.is-type equality

### ISSUE THREAD

**Issue by anematode** [mochajs/mocha#4648]
**Is your feature request related to a problem or a nice-to-have?? Please describe.**
I'm using Mocha on code involving numeric computations, which I expect to (correctly) return NaN sometimes. There is an `.is.NaN`, but this is a bit limiting as it does not work in deep equals, and doesn't really work in parallel with other tests (always have to use `.is.NaN` separately from other equality tests).

**Describe the solution you'd like**
I'd like a test that uses Object.is-style equality—that is, comparing NaNs as equal and signed zeros as different. A deep equals using Object.is equality would be even better.

**Describe alternatives you've considered**
In my current tests I use my own NaN-aware comparison code directly.

**Additional context**
If approved, I'd be willing to work on this feature myself! Also, apologies if this feature already exists and I overlooked it.

**Comment by juergba** [comment]
Mocha doesn't include its own assertions. Are you using Chai?

**Comment by anematode** [comment]
Oops.
